web/tests/stream_data.test.cjs

```diff
@@ -1537,3 +1537,33 @@ test("has_metadata_access", ({override}) => {
     social.subscribed = true;
     assert.equal(stream_data.has_metadata_access(social), true);
 });
+
+test("has_content_access", ({override_rewire}) => {
+    const social = {
+        subscribed: false,
+        color: "red",
+        name: "social",
+        stream_id: 2,
+        is_muted: false,
+        invite_only: false,
+        history_public_to_subscribers: false,
+        can_add_subscribers_group: nobody_group.id,
+        can_administer_channel_group: nobody_group.id,
+    };
+
+    override_rewire(stream_data, "has_metadata_access", () => false);
+    assert.equal(stream_data.has_content_access(social), false);
+
+    override_rewire(stream_data, "has_metadata_access", () => true);
```

Comment:
Ideally, we'd not rewire, but instead have everything here work through setting up the permissions objects.